platform/mbed_mem_trace.cpp

```diff
@@ -45,6 +46,18 @@ void mbed_mem_trace_set_callback(mbed_mem_trace_cb_t cb) {
     mem_trace_cb = cb;
 }
 
+void mbed_mem_trace_disable()
+{
+    mem_trace_cb_reserve = mem_trace_cb;
```

Comment:
Will it be interesting to check it in multi threaded system, when some thread trying to activate it and other disable it. I am not sure if that is a valid use case or not?

Reply:
@deepikabhavnani Updated it by adding the mutex locks used around malloc calls in the enable/disable functions. Do you think that'd be sufficient? Writing up a quick test in the meantime. Will squash the commits later if it works out fine.